topdown/walk.go

```diff
@@ -5,9 +5,32 @@
 package topdown
 
 import (
+	"sync"
+
 	"github.com/open-policy-agent/opa/ast"
 )
 
+var (
+	emptyArr = ast.ArrayTerm()
+
+	arrayPool = sync.Pool{
+		New: func() any {
+			return ast.NewArray(
+				// These values don't matter as they'll be immediately replaced.
+				// Used only to initialize the size of the array.
+				ast.InternedBooleanTerm(false),
```

Comment:
Can we pass `nil, nil` then? 🤔

Reply:
No, that'll be a nil dereference in the `ast.NewArray` function calling `.Value.Hash()` on each item.